muda/deformers/filter.py

```diff
@@ -0,0 +1,538 @@
+                value={
+                    "index": obs.value["index"],
+                    "frequency": new_freq,
+                    "voiced": voice,
+                },
+            )
+
+    @staticmethod
+    def filter_hz(annotation, state):
+        #non-existent pitch removed 
+        for obs in annotation.pop_data():
+            new_freq,voice = checkfreqinband(obs.value,state,datatype="hz")
+            if voice:
+                annotation.append(
+                    time=obs.time,
+                    duration=obs.duration,
+                    confidence=obs.confidence,
+                    value=new_freq,
+                )
+
+    @staticmethod
+    def filter_midi(annotation, state):
+         #non-existent pitch removed 
+        for obs in annotation.pop_data():
+            new_midi,voice = checkfreqinband(obs.value,state,datatype="midi")
+            if voice:
+                annotation.append(
+                    time=obs.time,
+                    duration=obs.duration,
+                    confidence=obs.confidence,
+                    value=new_midi,
+                )
+
+    @staticmethod
+    def filter_class(annotation, state):
+        #non-existent pitch removed 
+        for obs in annotation.pop_data():
+            value = deepcopy(obs.value)
+            
+            new_freq, voice = checkfreqinband(value["tonic"]+str(value["pitch"]),state,"pitchclass")
+            if voice:
+                value["tonic"] = new_freq[:-1]
+                value["pitch"] = int(new_freq[-1])
+                annotation.append(
+                time=obs.time,
+                duration=obs.duration,
+                confidence=obs.confidence,
+                value=value,
+            )
+            
+           
+
+
+class Filter(AbstractFilter):
+    """ Filtering by cheby2 iir filter
+    
+    This transformation affects the following attributes:
+
+    - Annotations 
+        - pitch_contour, pitch_hz, pitch_midi, pitch_class
+        - note_hz, note_midi
+    - Audio
+
+    Attributes
+    ----------
+    type: "low" or "high" or "bandpass"
+    order: int > 0
+        order of the filter
+    attenutation: float > 0
+        The minimum attenuation required in the stop band. 
+        Specified in decibels, as a positive number.
+
+    cutoff: in hz
+        can be float, list of float, or list of tuples in the case of bandpass filter
+
+    make one or more filters of the same type, but customized cutoff frequencies
+
+    See Also
+    --------
+    RandomFilter
+
+    Examples
+    --------
+    >>> # Shift down by a quarter-tone
+    >>> D = muda.deformers.Filter(btype,order,attenuation,cutoff)
+    """
+
+    def __init__(self,btype="low",order=4, attenuation=60.0, cutoff=4000):
+        AbstractFilter.__init__(self)
+        self.btype = btype
+        self.order = order
+        self.attenuation = attenuation
+        if self.btype == "bandpass":
+            if type(cutoff) == tuple:
```

Comment:
checking `type(x) == y` is an anti-pattern.

Instead, say `isinstance(cutoff, tuple)`.   (Similarly for following lines)